File: anthropic-proxy.js
import http from 'http';
import https from 'https';
import { URL } from 'url';

const server = http.createServer((req, res) => {
  // Set CORS headers
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type, x-api-key');

  if (req.method === 'OPTIONS') {
    res.writeHead(200);
    res.end();
    return;
  }

  // Handle models endpoint
  if (req.url === '/v1/models' && req.method === 'GET') {
    res.writeHead(200, { 'Content-Type': 'application/json' });
    res.end(JSON.stringify({
      object: "list",
      data: [
        { id: "claude-3-5-sonnet-20240620", object: "model", owned_by: "anthropic" },
        { id: "claude-3-5-haiku-20241022", object: "model", owned_by: "anthropic" },
        { id: "claude-3-opus-20240229", object: "model", owned_by: "anthropic" }
      ]
    }));
    return;
  }

  // Proxy all other requests to Anthropic
  const targetUrl = `https://api.anthropic.com${req.url}`;
  
  let body = '';
  req.on('data', chunk => {
    body += chunk.toString();
  });

  req.on('end', () => {
    const options = {
      hostname: 'api.anthropic.com',
      port: 443,
      path: req.url,
      method: req.method,
      headers: {
        'x-api-key': req.headers['x-api-key'],
        'anthropic-version': '2023-06-01',
        'content-type': 'application/json'
      }
    };

    const proxyReq = https.request(options, (proxyRes) => {
      res.writeHead(proxyRes.statusCode, proxyRes.headers);
      proxyRes.pipe(res);
    });

    proxyReq.on('error', (err) => {
      console.error('Proxy error:', err);
      res.writeHead(500);
      res.end(JSON.stringify({ error: 'Proxy error' }));
    });

    if (body) {
      proxyReq.write(body);
    }
    proxyReq.end();
  });
});

server.listen(3001, () => {
  console.log('Anthropic proxy running on http://localhost:3001');
});

File: anthropic-proxy.py
from flask import Flask, request, jsonify
import requests
import os

app = Flask(__name__)

@app.route('/v1/models', methods=['GET'])
def models():
    return jsonify({
        "object": "list",
        "data": [
            {"id": "claude-3-5-sonnet-20240620", "object": "model", "owned_by": "anthropic"},
            {"id": "claude-3-5-haiku-20241022", "object": "model", "owned_by": "anthropic"},
            {"id": "claude-3-opus-20240229", "object": "model", "owned_by": "anthropic"}
        ]
    })

@app.route('/v1/messages', methods=['POST'])
def messages():
    headers = {
        'x-api-key': request.headers.get('x-api-key'),
        'anthropic-version': '2023-06-01',
        'content-type': 'application/json'
    }
    
    response = requests.post(
        'https://api.anthropic.com/v1/messages',
        headers=headers,
        json=request.json
    )
    
    return response.text, response.status_code

@app.route('/health', methods=['GET'])
def health():
    return jsonify({"status": "ok"})

if __name__ == '__main__':
    port = int(os.environ.get('PORT', 3000))
    app.run(host='0.0.0.0', port=port)
